includes/panopto_derivatives.inc

```diff
@@ -5,37 +5,386 @@
+            'message_sub' => array(
+              '@endpoint' => $endpoint, 
+              '@expectation' => $expectation,
+              '@response' => $response,
+            ),
+            'type' => 'watchdog',
+            'severity' => WATCHDOG_ERROR,
+          );
+          $failureMessage['messages'][] = $messages;
+          watchdog($messages);
+        }
+      } else {
+        $messages = array(
+          'message' => t("Expected JSON key from @endpoint but could not decode. @response"),
+          'message_sub' => array(
+            '@endpoint' => $endpoint,
+            '@response' => $response,
+          ),
+          'type' => 'watchdog',
+          'severity' => WATCHDOG_ERROR,
+        );
+        $failureMessage['messages'][] = $messages;
+        watchdog($messages);
+      }
+    } else {
+      $messages = array(
+        'message' => t("Expected one of @expected; got a @response_code"),
+        'message_sub' => array(
+          '@expected' => implode(',', array_keys($expect)),
+          '@response_code' => $response_code,
+        ), 
+        'type' => 'watchdog',
+        'severity' => WATCHDOG_ERROR,
+      );
+      $failureMessage['messages'][] = $messages;
+      watchdog($messages);
+    }
+  }
+  return;
+}
+
+/**
+ * Uploads file to Panopto using AWS S3 Uploader
+ * @param $upload_target string The URL of upload target
+ * @param $file_path string The URL which is the target of the request
+ * @return void
+ */
+function islandora_video_streaming_panopto_upload_file($upload_target, $file_path) {
+  $element = explode("/", $upload_target);
+  $prefix = array_pop($element);
+  $service_endpoint = implode("/", $element);
+  $bucket = array_pop($element);
+  $file_name = basename($file_path);
+  $object_key = $prefix."/".$file_name;
+  
+  $s3Params = array(
+      'endpoint' => $service_endpoint,
+      'region'  => 'us-east-1',
+      'version' => '2006-03-01',
+      'credentials' => [
+        'key'    => 'dummy',
+        'secret' => 'dummy'
+      ]
+  );
+  // Create an S3Client
+  $s3Client = new S3Client($s3Params);
+
+  $source = fopen($file_path, 'rb');
+  $uploader = new ObjectUploader(
+    $s3Client,
+    $bucket,
+    $object_key,
+    $source
+  );
+
+  do {
+    try {
+      $result = $uploader->upload();
+    } catch (MultipartUploadException $e) {
+      rewind($source);
+      $uploader = new MultipartUploader($s3Client, $source, [
+        'state' => $e->getState(),
+      ]);
+    }
+  } while (!isset($result));
+}
+
+/**
+ * Creates manifest file, in temporary storage, needed for Panopto Video Upload
+ * @param $file_path The filepath of the Islandora object
+ * @return string The filename or filepath of the newly created manifest file
+ */
+function islandora_video_streaming_panopto_create_manifest_for_video($file_path = null) {
+  $file_name = basename($file_path);
+  $manifest_template = module_load_include('xml', 'islandora_video_streaming', 'includes/upload_manifest_template');
```

Comment:
`module_load_include` will treat this file as a PHP file, using `require`.   See: https://api.drupal.org/api/drupal/includes%21module.inc/function/module_load_include/7.x

Since this is a datafile you are going to reference, use something like `drupal_get_path` instead.  https://api.drupal.org/api/drupal/includes%21common.inc/function/drupal_get_path/7.x

Reply:
I'm having issues with drupal_get_path('xml','upload_manifest_template').

Reply:
> Parameters
> `$type`: The type of the item (i.e. theme, theme_engine, module, profile).
> `$name`: The name of the item for which the path is requested.

So, the path to your module would be:
`drupal_get_path('module','islandora_video_streaming')`

And the path to your XML could be:
`drupal_get_path('module','islandora_video_streaming') . PATH_SEPARATOR . 'includes' . PATH_SEPARATOR . 'upload_manifest_template.xml'`